subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy

```diff
@@ -87,9 +130,15 @@ class CreateFiles implements Runnable {
 
     @Override
     void run() {
+        if (Boolean.valueOf(System.getProperty("killMe"))) {
```

Comment:
we don't need the conditional logic, the build is always run this way

Reply:
It is not. The first run is executed without any killing.